cardano-c/src/key.rs

```diff
@@ -15,22 +15,49 @@ pub extern "C" fn cardano_xprv_derive(c_xprv: XPrvPtr, index: u32) -> XPrvPtr {
 }
 
 #[no_mangle]
-pub extern "C" fn cardano_xprv_from_bytes(c_xprv: *const u8) -> XPrvPtr {
+pub extern "C" fn cardano_xprv_from_bytes(
+    c_xprv: *const u8,
+    xprv_out: *mut XPrvPtr,
+) -> CardanoResult {
     let xprv_data = unsafe { slice::from_raw_parts(c_xprv, hdwallet::XPRV_SIZE) };
-    match hdwallet::XPrv::from_slice(xprv_data) {
+    let array = {
+        let mut array = [0u8; 96];
+        array.copy_from_slice(xprv_data);
+        array
+    };
+    match hdwallet::XPrv::from_bytes_verified(array) {
         Ok(r) => {
+            println!("ok");
```

Comment:
you might want to get rid of this debug message.

(btw, there is a `dbg!` macro you can use that are armless in release mode -- unlike a `println`)